CIP-0026/CIP-0026.md

```diff
@@ -0,0 +1,468 @@
+
+For example, the attestation message for the example entry above is:
+
+```js
+hash(
+  hash(0x75613534303864306462306439343266643830333734) +  // text "a5408d0db0d942fd80374"
+  hash(0x67636F6E74616374) +                              // text "contact"
+  hash(0x6A436964204B72616D6572) +                        // text "Cid Kramer"
+  hash(0x00)                                              // uint 0
+)
+// cd731afcc904c521e0c6b3cc0b560b8157ee29c3e41cd15f8dc8984edf600029
+```
+
+The `publicKey` and the `signature` MUST be base16-encoded and stored as JSON strings. All signatures must be verifiable through the **Ed25519 digital signature** scheme and public keys must therefore be 32-byte long Ed25519 public keys. 
+
+
+## Well-known Properties
+
+The following properties are considered well-known, and the JSON in their values MUST have the given structure and semantic interpretation. New properties can be added to this list by amending this CIP. The role of well-known properties is to facilitate integration between applications implementing this CIP. Nevertheless, registries are encouraged to **not** restrict properties to only this limited set but, registries (or metadata servers) MUST verify the well-formedness of those properties when present in a metadata object.
+
+<details>
+  <summary><code>name</code></summary>
+
+```json
+{
+    "type": "string",
+    "description": "A human-readable name for the metadata subject, suitable for use in an interface or in running text.",
+    "maxLength": 50,
+    "minLength": 1
+}
+```
+</details>
+
+
+<details>
+  <summary><code>description</code></summary>
+
+```json
+{
+    "type": "string",
+    "description": "A longer description of the metadata subject, suitable for use when inspecting the metadata subject itself.",
+    "maxLength": 500
+}
+```
+</details>
+
+<details>
+  <summary><code>ticker</code></summary>
+
+```json
+{
+  "type": "string",
+  "description": "A short identifier for the metadata subject, suitable to show in listings or tiles.",
+  "maxLength": 4,
+  "minLength": 2
+}
+```
+</details>
+
+<details>
+  <summary><code>decimals</code></summary>
+
+```json
+{
+  "type": "integer",
+  "description": "When the metadata subject refers to a monetary policy, refers to the number of decimals of the currency.",
+  "minimum": 0,
+  "maximum": 19
+},
+```
+</details>
+
+<details>
+  <summary><code>url</code></summary>
+
+```json
+{
+    "type": "string",
+    "description": "A universal resource identifier pointing to additional information about the metadata subject.",
+    "format": "uri",
+    "pattern": "/^https:/",
```

Comment:
Hmm. This is mostly matching the current design of the metadata server used by Daedalus and others.

 I'd say that, if we indeed limit this to HTTP URLs only, then either the property should be renamed to `httpUrl` altogether. The pattern restriction was mainly to restrict HTTP URLs to "secure URLs only". I don't think that a pattern listing all supported protocols would make a lot of sense, so perhaps it's okay to remove the pattern and leave it at the discretion of the server to accept or reject insecure URLs.

Reply:
Hmm, but should we allow, say... an IPFS URL? Generally I think it's okay to say this could be any kind of URL, but the client might have some restrictions on which ones it will open? Not sure.

Reply:
I guess if we're unsure of the policy it's probably better not to mandate something.